File: 0283-move-zeroes/0283-move-zeroes.js
var moveZeroes = function (nums) {
  let low = 0;
  let high = low + 1;
  for (let i = 0; i < nums.length - 1; i++) {
    if (nums[low] !== 0) {
      low++;
      high++;
    } else {
      if (nums[high] !== 0) {
        [nums[low], nums[high]] = [nums[high], nums[low]];
        low++;
      }
      high++;
    }
  }
  return nums;
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        s = f = 0
        while f < len(nums):
            if nums[f] != 0:
                nums[s] = nums[f]
                s += 1
            f += 1
        for i in range(s, len(nums)):
            nums[i] = 0
                